ebib-utils.el

```diff
@@ -2387,14 +2387,16 @@ are returned as a string."
 (defun ebib-display-www-link (field key db)
   "Return the content of FIELD from KEY in DB as a link.
 This function is mainly intended for the DOI and URL fields."
-  (let ((str (ebib-get-field-value field key db 'noerror 'unbraced 'xref)))
-    (if (and str (string-match-p "^[0-9]" str))
-        (setq str (concat "https://dx.doi.org/" str)))
-    (if str
-        (propertize "www"
-                    'face '(:height 0.8 :inherit ebib-link-face)
-                    'mouse-face 'highlight
-                    'help-echo str)
+  (let* ((val (ebib-get-field-value field key db 'noerror 'unbraced 'xref))
+	 ;; Unless field is doi, assume the string is a full url
+	 (str (if (case-fold-string= field "doi")
```

Comment:
Note that the DOI field may already start with `https://dx.doi.org/`, which is why the original code only adds it if the field value starts with a number. (DOIs tend to start with `10`, though I admit I'm not sure if that's mandatory. A better test would be to see if the field value starts with `https`.)

Reply:
Thanks for pointing this out. Fixed and pushed. Turns out, DOIs [*should* always start with `10`](https://www.doi.org/doi_handbook/2_Numbering.html#2.2.2), though I don't know whether this is actually enforced, so I've stuck with the check for a single leading number 0-9.